rq/worker.py

```diff
@@ -502,6 +481,21 @@ def work(self, burst=False, logging_level="INFO", date_format=DEFAULT_LOGGING_DA
 
                 except StopRequested:
                     break
+
+                except SystemExit:
+                    # Cold shutdown detected
+                    raise
+
+                except:  # noqa
+                    self.log.error(
+                        """
+                        Worker %s encountered an unhandled exception.\n
+                        Stacktrace:\n %s\n\n.
+                        Worker %s quitting...
+                        """,
+                        self.name, traceback.format_exc(), self.name
```

Comment:
maybe `exc_info=True` instead of including the traceback in the message?

Reply:
this is more canonical, and also works better on alternative log processors.

Reply:
Will try to update this.